code/modules/power/supermatter/supermatter.dm

```diff
@@ -304,6 +306,12 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			visible_message("<span class='warning'>[src] melts through [T]!</span>")
 		return
 
+	if(last_complete_process > SSair.last_complete_process) 
+		power_changes = FALSE //Atmos has not been fully processed since the previous time the SM was. Abort all power and processing operations.
```

Comment:
Will we even get here if atmos SS is not running?

Reply:
If it is completely off? No, we won't.
If it is chugging enough to cut off early? Yes, we will.

`SSair/fire()` does the following:
* check if a z-level has become active, and unpause the atmos there if so
* `Check if subsystem needs to pause`
* Check if z-levels have gone dormant and pause the atmos there if so
* `Check if subsystem needs to pause`
* Rebuild pipenets if necessary
* `Check if subsystem needs to pause`
* Process pipenets
* `Check if the subsystem has been set to pause`
* Process atmos machines **(this is where the SM is processed)**
* `Check if the subsystem has been set to pause`
* Check for and apply space wind
* `Check if the subsystem has been set to pause`
* Prepare turfs for processing atmos
* `Check if the subsystem has been set to pause`
* Generate or extinguish hotspots (flame tiles)  <- This is one of the locations it is very visible when atmos is freezing part-way through
* `Check if the subsystem has been set to pause`
* Process *some amount* of temperature exchange
* `Check if the subsystem has been set to pause`
* Process gas mixing between turfs <- Causes problems when the SM processes two or more ticks in a row without `fire()` making it to this point.
* End of process

Every chunk of atmos processing checks to see if it has run out of time in the current tick, and if it has, processing halts right there and the proc returns early. This means *any time* atmos is partially chugging, the actual gas mixing is not occurring even though some random amount of other procs are still going through. 

Z-level pausing and unpausing doesn't happen very often, relatively speaking, so the SM is usually the third step, but the part of the proc SM requires to appropriately process is the eighth and final step.

Reply:
Ah, how curious makes sense then

Reply:
It pauses and early returns, however when it fires() on the next run, resumed will be true which will prevent it from resetting its current processing queue. This means it will continue from where it left off last tick